pkg/client/config.go

```diff
@@ -23,32 +25,70 @@ type Config struct {
 
 // Auth describes the authentication information of a registry
 type Auth struct {
-	Username string `json:"username"`
-	Password string `json:"password"`
-	Insecure bool   `json:"insecure"`
+	Username string `json:"username" yaml:"username"`
+	Password string `json:"password" yaml:"password"`
+	Insecure bool   `json:"insecure" yaml:"insecure"`
 }
 
 // NewSyncConfig creates a Config struct
-func NewSyncConfig(configFilePath, defaultDestRegistry, defaultDestNamespace string) (*Config, error) {
-	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
-		return nil, fmt.Errorf("config file %v not exist: %v", configFilePath, err)
+// configFile
+func NewSyncConfig(configFile, authFilePath, imageFilePath, defaultDestRegistry, defaultDestNamespace string) (*Config, error) {
+	if len(configFile) == 0 && (len(authFilePath) == 0 || len(imageFilePath) == 0) {
+		return nil, fmt.Errorf("need to supply config.json or supply both auth.json and images.json")
 	}
 
-	file, err := os.OpenFile(configFilePath, os.O_RDONLY, 0666)
-	if err != nil {
-		return nil, fmt.Errorf("open config file %v error: %v", configFilePath, err)
+	var config Config
+
+	if len(configFile) != 0 {
+		if err := openAndDecode(configFile, &config); err != nil {
+			return nil, fmt.Errorf("decode config file %v failed, error %v", configFile, err)
+		}
+	} else {
+		if err := openAndDecode(authFilePath, &config.AuthList); err != nil {
+			return nil, fmt.Errorf("decode auth file %v error: %v", authFilePath, err)
+		}
+
+		if err := openAndDecode(imageFilePath, &config.ImageList); err != nil {
```

Comment:
I have a little problem here, is it possible that only a "images" config file is needed for the people that do not need authentication ?

Reply:
I can't think of any repositories that can be pushed images without authentication. But I'll add it in case of there is one.